### PR TITLE
cluster templates details page

### PR DESCRIPTION
**Pull request by batzionb**
Signed-off-by: Bat Zion Rotman <brotman@redhat.com>

### CODE REVIEW

**Comment by batzionb** [comment]
image

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+jest.mock('../hooks/useClusterTemplateInstances', () => {
+  return {
+    useClusterTemplateInstances: () => [[], true, null],
+  };
+});
+
+jest.mock('@openshift-console/dynamic-plugin-sdk', () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  return {
+    ResourceLink: jest.fn().mockImplementation(({ name }) => <div>{name}</div>),
```
Comment (jtomasek) [inline frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx:25]: Alternatively you can use `MockComponent`

```suggestion
  const MockComponent = require('../mocks/MockComponent').default;
  return {
    ResourceLink: MockComponent,
```

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsSections.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+const ClusterTemplateDetailsSections: React.FC<{ clusterTemplate: ClusterTemplate }> = ({
+  clusterTemplate,
+}) => {
+  const { t } = useTranslation();
+  const titles = getSectionTitles(t);
+  const components = getSectionComponents();
+  return (
+    <Stack hasGutter>
+      {Object.keys(titles).map((key) => {
+        const section: Section = key as any as Section;
```
Comment (jtomasek) [inline frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsSections.tsx:44]:
```suggestion
        const section = key as Section;
```

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/QuotasSection.spec.tsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+const quotasInTemplate = quotas.filter((quota) =>
+  (quota.spec?.allowedTemplates?.map((templateData) => templateData.name) || []).includes(
+    clusterTemplate.metadata.name,
+  ),
+);
+
+jest.mock('@openshift-console/dynamic-plugin-sdk', () => {
+  // eslint-disable-next-line @typescript-eslint/no-var-requires
+  return {
+    ResourceLink: jest.fn().mockImplementation(({ name }) => <div id="resource-name">{name}</div>),
```
Comment (jtomasek) [inline frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/QuotasSection.spec.tsx:23]: Could use `MockComponent` as mentioned above.

**Review by jtomasek — CHANGES_REQUESTED** [review]

File: frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx (hunk 1 above)
Comment (batzionb) [reply frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx:25]: The idea was that mocking ResourceLink this way I can verify that the name was sent correctly, by later verifying the text content of the mocked component

Comment (batzionb) [reply frontend/src/routes/Infrastructure/ClusterTemplates/ClusterTemplateDetails/ClusterTemplateDetailsPage.spec.tsx]: In the end I figured out how to use it

File: frontend/src/routes/Infrastructure/ClusterTemplates/helpers/TableLoader.tsx [hunk 4]
```diff
@@ -47,13 +39,12 @@ function TableLoader({
...
-          {errorTitle ?? t('Something went wrong')}
-        </Title>
-        {errorMessage && <EmptyStateBody>{errorMessage}</EmptyStateBody>}
-      </EmptyState>
+      <ErrorState
+        error={error}
+        errorId={errorId}
+        errorTitle={errorTitle}
+        errorMessage={errorMessage}
+      ></ErrorState>
```
Comment (jtomasek) [inline frontend/src/routes/Infrastructure/ClusterTemplates/helpers/TableLoader.tsx:47]:
```suggestion
      />
```

**Review by jtomasek — APPROVED** [review]

**Pull request merged**